### DISCUSSION TITLE
Arg regex in Field Class not work, and  Type Literal not supported

### DISCUSSION THREAD

**Discussion by leokyler** [fastapi/sqlmodel#1259] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated, Literal
from sqlmodel import Field, SQLModel
from pydantic import AfterValidator

ROLES_SET = {"student", "teacher", "admin"}


def check_role(x: str):
    assert x in ROLES_SET, f"role must be in {'student', 'teacher', 'admin'}"
    return x

class UserBase(SQLModel):
    role: Annotated[str, AfterValidator(check_role)]  # worked
    # role: Literal["student", "teacher", "admin"] # when use it with fastapi,the User class will trigger error.
    # role: Annotated[str, Field(regex=r'^(student|teacher|admin)$')]  #no error,but not work,and string can pass

class User(UserBase, table=True):
    id: int | None = Field(default=None, primary_key=True)
```

### Description

In my code, I write two class extend from SQLmodel.

Attr 'role'（str） in class should in  ["student", "teacher", "admin"].
In the normal  python /pydantic class system, I can use `role: Literal["student", "teacher", "admin"]`。
But when I set it in a class extend SQLmodel, I get error:

File "/home/leo/myspace/collaborative_learning_system/server/models/tables.py", line 76, in <module>
    class User(UserBase, table=True):
  File "/home/leo/myspace/collaborative_learning_system/server/venv/lib/python3.12/site-packages/sqlmodel/main.py", line 559, in __new__
    col = get_column_from_field(v)
          ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/leo/myspace/collaborative_learning_system/server/venv/lib/python3.12/site-packages/sqlmodel/main.py", line 708, in get_column_from_field
    sa_type = get_sqlalchemy_type(field)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/leo/myspace/collaborative_learning_system/server/venv/lib/python3.12/site-packages/sqlmodel/main.py", line 656, in get_sqlalchemy_type
    if issubclass(type_, Enum):
       ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: issubclass() arg 1 must be a class

As a substitute, I think maybe can use `role: Annotated[str, Field(regex=r'^(student|teacher|admin)$')] `,but when I use it with fastapi, I found any string can be inserted into the database(postgresql). 
In addition,I found regex in the Field from pydantic can work! If use it, the code like: 
`role: Annotated[str, Field(partten=r'^(student|teacher|admin)$')]`

Final, I have to use:  `Annotated[str, AfterValidator(check_role)]`, it can work ,but not simple and not grace.

### Operating System

Linux

### Operating System Details

ubuntu23 (wsl)

### SQLModel Version

0.0.22

### Python Version

python3.12.3

### Additional Context

![image](https://github.com/user-attachments/assets/1d034f2d-e49b-4402-bd14-669fa976572d)

**Comment by YuriiMotov** [comment]
This is a duplicate of https://github.com/fastapi/sqlmodel/discussions/735 (`regex`) and https://github.com/fastapi/sqlmodel/issues/57 (`Literal`).

I'll close this discussion. Let's track these issues there
